pandas/io/json/json.py

```diff
@@ -129,6 +129,10 @@ def _write(self, obj, orient, double_precision, ensure_ascii,
                date_unit, iso_dates, default_handler):
         if not self.index and orient == 'split':
             obj = {"name": obj.name, "data": obj.values}
+        if orient == 'records':
+            if not obj.name:
+                obj.name = 0
+            obj = [{obj.name: v} for v in obj.values]
```

Comment:
Hmm not sure about the implementation here and accessing `.values`. Did you have any alternate implementations you were considering? Wondering if this shouldn't even be handled a level deeper, maybe even as far back as the actual encoder:

https://github.com/pandas-dev/pandas/blob/b9ddd7ad95c79ae10a5581fd13ff7260643311b6/pandas/_libs/src/ujson/python/objToJSON.c#L2084

Reply:
i tried a bit, and changed to another implementation. Look forward to your opinion!